server/src/main/java/org/elasticsearch/common/util/concurrent/TaskExecutionTimeTrackingEsThreadPoolExecutor.java

```diff
@@ -15,20 +15,22 @@
 import java.util.concurrent.RejectedExecutionHandler;
 import java.util.concurrent.ThreadFactory;
 import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.LongAdder;
 import java.util.function.Function;
 
 /**
- * An extension to thread pool executor, which tracks the exponentially weighted moving average of the task execution time.
+ * An extension to thread pool executor, which tracks statistics for the task execution time.
  */
-public final class EWMATrackingEsThreadPoolExecutor extends EsThreadPoolExecutor {
+public final class TaskExecutionTimeTrackingEsThreadPoolExecutor extends EsThreadPoolExecutor {
```

Comment:
The names a bit verbose/long, but we rarely use it directly. I'm open for suggestions.

Reply:
It follows the venerable Java naming tradition, it's good 😅